Makefile

```diff
@@ -2,24 +2,66 @@
 # MAINTAINER: Steve Sloka <slokas@upmc.edu>
 # If you update this image please bump the tag value before pushing.
 
-.PHONY: all binary container push clean test
-
 TAG = 1.6
 PREFIX = upmcenterprises
 
+BIN = registry-creds
+ARCH = amd64
+
+# go option
+GO      ?= go
+LDFLAGS := -w
+GOFLAGS := -a -installsuffix cgo
+
+# docker build arguments for internal proxy
+ifneq ($(http_proxy),)
+HTTP_PROXY_BUILD_ARG=--build-arg http_proxy=$(http_proxy)
+else
+HTTP_PROXY_BUILD_ARG=
+endif
+
+ifneq ($(https_proxy),)
+HTTPS_PROXY_BUILD_ARG=--build-arg https_proxy=$(https_proxy)
+else
+HTTPS_PROXY_BUILD_ARG=
+endif
+
+.PHONY: all
 all: container
 
-build: main.go
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -o registry-creds --ldflags '-w' ./main.go
+.PHONY: build
+build: build-dirs bin/$(BIN) bin/linux_$(ARCH)/$(BIN)
+
+# local developement binary (auto detect developer OS)
+bin/$(BIN): main.go
```

Comment:
Why `go install` each time? If I'm just building, I don't see the need to put into path.